File: src/main/java/com/kn/knwremodel/Comment/CommentRepository.java
package com.kn.knwremodel.Comment;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CommentRepository extends JpaRepository<Comment, Long> {

    @Query("SELECT c FROM Comment c WHERE c.user.id = :userId")
    List<Comment> findCommentsByUserId(@Param("userId") Long userId);
    
}

File: src/main/kotlin/injea/knwremodel/Comment/CommentRepository.kt
package injea.knwremodel.Comment

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface CommentRepository : JpaRepository<Comment?, Long?> {
    @Query("SELECT c FROM Comment c WHERE c.user.id = :userId")
    fun findCommentsByUserId(@Param("userId") userId: Long?): List<Comment?>?
}

